### PR TITLE
2.0.0

### PR DESCRIPTION
**Pull request by holl-**
Φ-Flow version 2 is the first major update since public release.

It introduces a functional style for writing simulations and provides lots of utility to go along with it. The most fundamental addition is a new `Tensor` class that wraps native tensors and extends them by named and typed dimensions. This makes code simulation more generic and robust, and less prone to errors.

### New major features

*General & UI*

* API documentation with pdoc
* phi.verify() and verify.py
* Matplotlib UI
* Overhauled UI API

*Physics & Fields*

* Easy visualization of grids and point clouds using `vis.plot()`
* FLIP simulation functions and demo
* Improved PointCloud support
* Field I/O, batched Scenes
* Unified SampledFields

*Math*

* New classes `Tensor`, `Shape` and `TensorLike` classes
* Full support for PyTorch, TensorFlow 2 and Jax
* Unified gradient computation (`functional_gradient()`, `custom_gradient`)
* Unified just-in-time compilation (`jit_compile()`)
* Profiling
* TensorFlow 2 with eager execution
* Named and typed dimensions for automatic vectorization, reshaping and easy dimensionality scaling
* Automatic conversion from NumPy
* Automatic dtype casting
* Automatic sparse matrix generation for linear functions
* Non-uniform tensors
* Lazy allocation and stacking
* Control FP precision globally or via context managers
* Unified extrapolations
* DLPack conversion with `convert`
* Integration of SciPy optimizers

### Release Candiates

RC0: #27
RC1: #29
RC2: #31

### CODE REVIEW

**Review by thunil — APPROVED** [review]
Looks good to me
